core/src/main/java/org/frankframework/stream/Message.java

```diff
@@ -187,6 +195,22 @@ public MessageContext copyContext() {
 		return new MessageContext(getContext());
 	}
 
+	private static class MessageNotClosedAction implements Runnable {
+		private boolean calledByClose = false;
+		private final String content;
+
+		private MessageNotClosedAction(String content) {
+			this.content = content;
+		}
+
+		@Override
+		public void run() {
+			if (!calledByClose) {
+				LOG.info("Leak detection: Message was not closed properly! Content: [{}]", content);
+			}
+		}
+	}
+
```

Comment:
> So we use this purely for leak-detection here and not for any actual close-actions.
> 
> Shouldn't we really do the full `close()` here?

NB: Thinking more about the state of the code, if we do actually run the full close from the cleanup-action of Message we might quickly run into a lot of problems due to code such as in `public static InputStream Message.asInputStream(Object object) { ... }` which creates a message and then discards it immediately. If the `Object` was any kind of `AutoClosable` this will instantly cause problems with current code base so perhaps we should leave it at a warning for now.

Reply:
Yes, it might be interesting to see the SFR and GraphvizEngine working and see how often the Cleaner is logging and doing its job. 
Secondly, for Message, in the current close-method it checks the `request` and `resourcesToClose`. When moving that to the Cleaner, we need to keep or get the reference in the cleaner class as well.. Not sure if that will actually work, since it might screw up the system because it is keeping references. Anyways, I'll leave it for now.

Reply:
> Secondly, for Message, in the current close-method it checks the `request` and `resourcesToClose`. When moving that to the Cleaner, we need to keep or get the reference in the cleaner class as well.. Not sure if that will actually work, since it might screw up the system because it is keeping references. Anyways, I'll leave it for now.

Yes that is a good point. Of course it should not be a huge deal in principle, but for now let's be safe and not do that yet.

Reply:
PS; De method in Message die ik bedoelde is trouwens `public static InputSource asInputSource(Object object) throws IOException`, niet `asInputStream(Object)`. En blijkt de enige te zijn die op deze manier enge dingen doet. Dus misschien valt het mee.